### PR TITLE
chore: add typescript checking

### PR DESCRIPTION
**Pull request by alvaroraminelli**
**What changed? Why?**

Improve the DevX allowing developer check typescript issues before opening a PR.

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: .github/workflows/app.js.yml [hunk 1]
```diff
@@ -31,6 +31,9 @@ jobs:
       - name: App Format Check
         working-directory: ./template
         run: npm run format:check
+      - name: App Typescript Check
+        working-directory: ./template
+        run: npm run typecheck
```
Comment (Zizzamia) [inline .github/workflows/app.js.yml:36]: Isn't this already being run under `yarn lint`?

@wespickett thought on this?

File: template/package.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
     "format:check": "prettier --check .",
     "lint": "next lint --fix",
     "lint:check": "next lint",
+    "typecheck": "tsc --noEmit",
```
Comment (Zizzamia) [inline template/package.json:13]: Why? should this part of `yarn lint`?

cc @wespickett

File: .github/workflows/app.js.yml (hunk 1 above)
Comment (Zizzamia) [reply .github/workflows/app.js.yml:36]: Ok dope, so if that's the case we should have the `yarn check` command, use `typecheck` as well.

**Comment by arsood** [comment]
Closing for now. Can reopen if necessary.

**Pull request closed without merging**